### DISCUSSION TITLE
Better Linux Autocompletion

### DISCUSSION THREAD

**Discussion by ChrisYeomans** [vlang/v#11589] [Ideas]
At the moment when I use tab auto completion in the bash shell for ```v run start_of_name <tab>``` it will select a binary file, i.e. a file with no extension before it will select a .v file.
Every other compiler I have come across will select a file with the appropriate extension when doing this, could this be added into v?
Example:
My directory looks like this:
```
test.v test
```
If I enter ```v run t <tab>```, I will get ``` v run test```. What I want to get is ```v run test.v```

**Comment by silversquirl** [comment]
This is a Bash thing, not a Linux-specific thing.
[bash-completion](https://github.com/scop/bash-completion) may be a better place for completion scripts, as opposed to storing them in v's repo.

**Comment by crthpl** [comment]
can this be closed?

**Comment by larpon** [comment]
This is a valid request and should stay open IMO.

[V's Bash completion](https://github.com/vlang/v/blob/639061be6ce13e3ab3fbd1f8c73c284c9c58a90b/cmd/tools/vcomplete.v#L306) can be modified to support this.

In contray to what @vktec says, you should note that I made the implementation so V has as much control over the completions as possible in order to be able to support as many shells as possible, with *as little interference* from the shells as possible.

**Comment by JalonSolov** [comment]
This may simply be a case of "not tested recently".  Note the last real update (from vktec) was in mid-2019.  Long before your completion stuff was done, Larpon.

**Comment by larpon** [comment]
@JalonSolov oh my you're right, it's from before we even had the internal autocompletions 👀

@vktec is right then, my bad, sorry 😬
